fdbserver/DataDistributionTracker.actor.cpp

```diff
@@ -439,15 +457,17 @@ Future<Void> shardMerger(
 		}
 
 		merged = KeyRangeRef( prevIter->range().begin, nextIter->range().end );
-		endingStats += newMetrics.get();
+		endingStats += newMetrics.get().metrics;
+		lastLowBandwidthTime = newMetrics.get().lastLowBandwidthTime;
 		if((forwardComplete ? prevIter->range().begin : nextIter->range().begin) >= systemKeys.begin) {
-			systemBytes += newMetrics.get().bytes;
+			systemBytes += newMetrics.get().metrics.bytes;
 		}
 		shardsMerged++;
 
 		auto shardBounds = getShardSizeBounds( merged, maxShardSize );
 		if( endingStats.bytes >= shardBounds.min.bytes ||
 				getBandwidthStatus( endingStats ) != BandwidthStatusLow ||
+				now() - lastLowBandwidthTime < SERVER_KNOBS->DD_LOW_BANDWIDTH_DELAY ||
```

Comment:
I understand and agree we should not merge a shard unless it has been in lowBandwidth status for a while.

This change will increase the number of shards. Do we have a rough estimation on the extra number of shards this change may potentially increase? (The `DD_LOW_BANDWIDTH_DELAY ` value may affect the number.)

Is the current number of shards close to the maximum number of shards a cluster can tolerate without experiencing performance issues?

Reply:
The effect this will have on the number of shards is heavily dependent on the client workload. There is very little risk that this will increase the shard count enough to impact the database performance, but we should monitor the number of shards after we roll out this change.